include/cudaq/Optimizer/Transforms/Passes.h

```diff
@@ -58,6 +58,13 @@ std::unique_ptr<mlir::Pass>
 createArgumentSynthesisPass(mlir::ArrayRef<mlir::StringRef> funcNames,
                             mlir::ArrayRef<mlir::StringRef> substitutions);
 
+/// Helper function to build an argument synthesis pass. The names of the
+/// functions and the substitutions text can be built as an unzipped pair of
+/// lists.
+std::unique_ptr<mlir::Pass>
+createArgumentSynthesisPass(const std::vector<std::string> &funcNames,
+                            const std::vector<std::string> &substitutions);
```

Comment:
Why do we need this? `ArrayRef` should subsume rigid `std::vector` here. I don't think we need this overload?

Reply:
`std::vector<string>` does not get auto-converted to `ArrayRef<StringRef>`...  I can try `ArrayRef<std::string>` instead